docs/sources/flow/reference/components/module.string.md

```diff
@@ -0,0 +1,135 @@
+`content` is typically loaded via the exports of another component. For example,
+
+- local.file.LABEL.content
+- remote.http.LABEL.content
+- remote.s3.LABEL.content
+
+`arguments` allows us to pass parameterized input into a module.
+An `argument` marked non-optional in the module being loaded is required in the
+`arguments`. It is also not valid to provide an `argument` not defined in the
+module being loaded.
+
+*TODO: Add link to argument config-blocks page once merged*
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`exports` | `map(any)` | The exports of the Module loader.
+
+`exports` exposes the `export` config block inside a module. It can be accessed from
+the parent river config via `module.string.LABEL.exports.EXPORT_LABEL
+
+*TODO: Add link to export config-blocks page once merged*
+
+## Component health
+
+`module.string` is reported as healthy if the most recent load of the module was 
+successful. 
+
+If the module is not loaded successfully, the current health displays as
+unhealthy and the health includes the error from loading the module.
+
+## Debug information
+
+`module.string` does not expose any component-specific debug information.
+
+### Debug metrics
+
+`module.string` does not expose any component-specific debug metrics.
+
+## Example
+
+In this example, we pass credentials from a parent river config to a module for
+a `prometheus.remote_write` flow component. We then export that component for use
+in the parent river config. The export can then be used by the parent river config
+as a component.
+
+Parent:
+
+```river
+local.file "metrics" {
+	filename = "/path/to/prometheus_remote_write_module.river"
+}
+
+module.string "metrics" {
+	content   = local.file.metrics.content
+	arguments = {
+		username = env("PROMETHEUS_USERNAME"),
+		password = env("PROMETHEUS_PASSWORD"),
+	}
+}
+
+prometheus.exporter.unix { }
+
+prometheus.scrape "local_agent" {
+	targets         = prometheus.exporter.unix.targets
+	forward_to      = [module.string.metrics.exports.prometheus_remote_write.receiver]
+	scrape_interval = "10s"
+}
+```
+
+Module:
+
+```river
+argument "username" { }
+
+argument "password" { }
+
+export "prometheus_remote_write" {
+	value = prometheus.remote_write.grafana_cloud
+}
+
+prometheus.remote_write "grafana_cloud" {
+	endpoint {
+		url = "https://prometheus-us-central1.grafana.net/api/prom/push"
+
+		basic_auth {
+			username = argument.username.value
+			password = argument.password.value
+		}
+	}
+}
+```
+
+> **NOTE**: For the purpose of the above example we have exported the entire
+> `prometheus.remote_write.grafana_cloud` component. Alternatively, the module above
+> could have exported `prometheus.remote_write.grafana_cloud.receiver` instead of
+> the entire component since that is the only part of the component we need in the
+> parent river config.
```

Comment:
I think we can drop this note, or have it only export the `receiver` if that will be the most common thing to do.

Reply:
I can drop the note. I think most common will be people exporting full components but we will see!